src/crafting_gui.cpp

```diff
@@ -1415,7 +1538,7 @@ const recipe *select_crafting_recipe( int &batch_size_out, const recipe_id goto_
             user_moved_line = highlight_unread_recipes;
         } else if( action == "CONFIRM" ) {
             if( available.empty() || !available[line].can_craft ) {
-                popup( _( "You can't do that!  Press [<color_yellow>ESC</color>]!" ) );
+                popup_getkey( _( "You can't do that!  Press [<color_yellow>ESC</color>]!" ) );
```

Comment:
Why is this changed?

Reply:
When I tested it, the original popup would only close with `Escape` or equivalent, while popup_getkey will close with a mouse click.  So, if someone is primarily using the mouse and double-clicks on a recipe that they can't make, the only way to get out of the popup() is to reach over to the keyboard.  It's a minor thing, but it's something that I personally would find annoying.

If the use of popup_getkey stands, the text should probably be updated to 'press any key' or something like that.

Reply:
You could probably use a `query_popup` with `allow_cancel` and a single `QUIT` option, which has been recently changed to support mouse click if I remember correctly.